File: Class 2/Problem 27/removeResult.js
let nums = [3, 2, 2, 3];
let val = 3;

var removeElement = function (nums, val) {
  let r = 0;
  let i = 0;

  while (i < nums.length) {
    if (nums[i] != val) {
      nums[r] = nums[i];
      r += 1;
    }
    i += 1;
  }
  return r;
};

console.log(removeElement(nums, val));

File: Class 2/Problem 27/removeResult.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        
        r = 0
        i = 0  
        while i < len(nums):
            if nums[i] != val:
                nums[r] = nums[i]
                r+=1
            i+=1

        print(nums)

        return len(nums[0:r])